MM2RandoLib/Random/DefaultSeed.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using MM2Randomizer.Extensions;
+
+namespace MM2Randomizer.Random
+{
+    /// <summary>
+    /// Provides basic randomization methods
+    /// </summary>
+    public sealed class DefaultSeed : ISeed
+    {
+        //
+        // Constructors
+        //
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("CodeQuality", "IDE0051:Remove unused private members", Justification = "Private constructors are used by the SeedFactory.")]
+        private DefaultSeed(String in_SeedString)
+        {
+            if (null == in_SeedString)
+            {
+                throw new ArgumentNullException(nameof(in_SeedString));
+            }
+
+            Int32 int32Seed = in_SeedString.ToInt32Hash();
+            String alpha26Seed = int32Seed.ToAlphaBase26();
+            System.Random random = new System.Random(int32Seed);
+
+            this.mRandom = random;
+            this.mSeedAlphaBase26 = alpha26Seed;
+            this.mSeedString = in_SeedString;
+        }
+
+        private DefaultSeed()
+        {
+            // Get a random number for the seed
+            System.Random r = new System.Random();
+            StringBuilder sb = new StringBuilder();
+
+            const Int32 SEED_STRING_LENGTH = 20;
+            const String SEED_STRING_CHARACTERS = "1234567890";
+
+            for (Int32 index = 0; index < SEED_STRING_LENGTH; ++index)
+            {
+                sb.Append(SEED_STRING_CHARACTERS[r.Next(0, SEED_STRING_CHARACTERS.Length)]);
+            }
+
+            String seedString = sb.ToString();
+
+            Int32 int32Seed = seedString.ToInt32Hash();
+            String alpha26Seed = int32Seed.ToAlphaBase26();
+            System.Random random = new System.Random(int32Seed);
+
+            this.mRandom = random;
+            this.mSeedAlphaBase26 = alpha26Seed;
+            this.mSeedString = seedString;
+        }
+
+
+        //
+        // Properties
+        //
+
+        public String SeedString
+        {
+            get
+            {
+                return this.mSeedString;
+            }
+        }
+
+
+        public String Identifier
+        {
+            get
+            {
+                return this.mSeedAlphaBase26;
+            }
+        }
+
+
+        //
+        // Public Methods
+        //
+
+        public void Next()
+        {
+            this.mRandom.Next();
+        }
+
+        // Boolean Methods
+        public Boolean GetNextBoolean()
```

Comment:
After reading through the code that uses ISeed, I think prefixing these next functions with `Get` is dubious. I expect calling a getter to be idempotent. That is, I expect it to return the same value until the object changes state. But these `GetNextFoo()` methods mutate the object and return a new thing. So I think `NextFoo()` is a better name.

Reply:
Hmm, that is a good point.  I'll make this change.

Reply:
Fixed!